clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1511,6 +1511,32 @@ public void seek(TopicPartition partition, long offset) {
         try {
             log.debug("Seeking to offset {} for partition {}", offset, partition);
             this.subscriptions.seek(partition, offset);
+            this.metadata.resetLastSeenEpoch(partition);
```

Comment:
Is this necessary? Without additional information, it seems reasonable to keep the last epoch we have seen. Same in `seekToBeginning` and `seekToEnd`.

Reply:
Makes sense. Thanks 👍